jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt

```diff
@@ -420,6 +439,16 @@ class CodeWhispererPopupManager {
         editor.selectionModel.addSelectionListener(codewhispererSelectionListener)
         Disposer.register(states) { editor.selectionModel.removeSelectionListener(codewhispererSelectionListener) }
 
+        val codewhispererDocumentListener: DocumentListener = object : DocumentListener {
```

Comment:
@andrewyuq, I'm wondering if we can put this logic inside https://github.com/aws/aws-toolkit-jetbrains/blob/main/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/editor/CodeWhispererEditorListener.kt#L13.

Reply:
They are different document listeners, that document listener should be attached as soon as editor is created, but for this one it should only be attached when CodeWhisperer's popup is about to show. (i.e. listener's scope is limited to `states` which are just when CodeWhisperer's section is active.